### ISSUE TITLE
BOM encoding in documentation

### ISSUE THREAD

**Issue by HedvigS** [tidyverse/readr#1187]
I'm struggling with reading in files with encoding UTF-8-BOM with readr functions. I see different discussions of this in different places, and it'd be really great if it was clarified in [the readr documentation on CRAN ](https://cran.r-project.org/web/packages/readr/readr.pdf) what the intended set-up is.

Expected behaviour: specify "UTF-8_BOM" as encoding argument

```r
library(readr)
df <-  read_tsv("tsv_file_with_BOM.tsv, encoding = "UTF-8-BOM")
```
result

```
Error in read_tsv(fn, encoding = "UTF-8-BOM") : 
  unused argument (encoding = "UTF-8-BOM")
```

I see that for `write_excel_csv2()` the documentation specifies that this has implications for BOM, it'd be nice if the documentation also specifies how to deal with this when reading such files in.

**Comment by jimhester** [comment]
You don't need to specify anything explicitly when reading in a byte order mark, readr automatically skips this when reading.

**Comment by HedvigS** [comment]
Oh? 

Hm, I'm having parsing problems with files with BOM, things aren't aligning correctly when I use `readr::read_tsv() `but they are with `read.delim(x, fileEncoding = "UTF-8-BOM", sep = "\t")`. I'll try and see if I can whip up any example files.

**Comment by jimhester** [comment]
Fixed by #1172, the first edition parser of readr only looked for a BOM in the CSV parser, not the TSV parser, but the second edition removes it in all cases.
